### PR TITLE
feat(HIS): introduction to the HIS checker

### PR DESCRIPTION
**Pull request by AfonsoSantos96**
This is the introduction PR to add the HIS metric checker to the bao-ci workflow.
The HIS metrics define recommended thresholds for a set of key code quality metrics (explicit in afterward PRs).
This PR still does not include the checker script itself but prepares the CI structure to eventually run it.

### Type of change

- **feat**: introduces a new functionality
  - Logical unit: his_checker

## Checklist:

- [ ] The changes generate no new warnings when building the project. If so, I have justified above.
- [ ] I have run the CI checkers before submitting the PR to avoid unnecessary runs of the workflow.

### CODE REVIEW

**Review by danielRep — CHANGES_REQUESTED** [review]
Just tried with bao-partitioner. LGTM, however you still need to commit the `his_checker.py` with `x` permissions.

**Comment by AfonsoSantos96** [comment]
> Just tried with bao-partitioner. LGTM, however you still need to commit the `his_checker.py` with `x` permissions.

Sure! Just changed the permissions in the last commit

File: his_checker.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+if __name__ == "__main__":
+    PARSER = argparse.ArgumentParser()
+    PARSER.add_argument("files", nargs="+", help="The files to process")
+    ARGS = PARSER.parse_args()
+
+    process_metric(ARGS)
+
+    if METRIC_FAIL:
+        sys.exit(-1)
```
Comment (josecm) [inline his_checker.py:30]: Why not have a metric function array and then a cycle through it.

```suggestion
   metrics = [ ]
   for process_metric in metrics:
       process_metric(ARGS)
       if METRIC_FAIL:
           sys.exit(-1)
```

Then, in the future, for each metric's processing function, just add i to the array.

**Review by josecm — APPROVED** [review]

File: his_checker.py (hunk 1 above)
Comment (AfonsoSantos96) [reply his_checker.py]: That is a more flexible way to implement the checker flow. I just followed your idea, thanks!

**Pull request merged**